### ISSUE TITLE
download file from remote error with compress

### ISSUE THREAD

**Issue by wangshuang0220** [rclone/rclone#4970]
#### What is the problem you are having with rclone?

when copy remote files to local with compress,then met some error as below:
Failed to copy with 6 errors: last error was: multipart copy: read failed: flate: corrupt input before offset 3607

ps:Maybe it has something to do with compressing the uploaded files
#### What is your rclone version (output from `rclone version`)

rclone v1.54.0-beta.5074.eb090d354

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

CentOS Linux release 7.6.1810 (Core)

####  Which cloud storage system are you using? (e.g. Google Drive)

S3

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

rclone copy compress:test /tmp/

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

2021/01/27 21:35:07 DEBUG : shard_mongo-kkpntt9grhdj_shard_group1-data/index-23-5012280989111803931.wt: multi-thread copy: stream 1/2 failed: context canceled
2021/01/27 21:35:07 ERROR : shard_mongo-kkpntt9grhdj_shard_group1-data/index-23-5012280989111803931.wt: Failed to copy: multipart copy: read failed: flate: corrupt input before offset 3607
2021/01/27 21:35:07 ERROR : Attempt 3/3 failed with 6 errors and: multipart copy: read failed: flate: corrupt input before offset 3607
2021/01/27 21:35:07 INFO  : 
Transferred:   	   66.188M / 66.188 MBytes, 100%, 2.465 MBytes/s, ETA 0s
Errors:                 6 (retrying may help)
Checks:               254 / 254, 100%
Transferred:          121 / 121, 100%
Elapsed time:        27.1s

2021/01/27 21:35:07 DEBUG : 4 go routines active
2021/01/27 21:35:07 Failed to copy with 6 errors: last error was: multipart copy: read failed: flate: corrupt input before offset 3607

tks

**Comment by ncw** [comment]
It looks like the compress backend doesn't like multithread copy...

@wangshuang0220 if you retry with `--multi-thread-streams 0` does it work properly.

@buengese compress supports seeking doesn't it so this should work I would have thought?

**Comment by wangshuang0220** [comment]
> It looks like the compress backend doesn't like multithread copy...
> 
> @wangshuang0220 if you retry with `--multi-thread-streams 0` does it work properly.
> 
> @buengese compress supports seeking doesn't it so this should work I would have thought?

i have tried with   `--multi-thread-streams 0` , it seems work . Is there any other way？Will the 1.54 release release solve this problem？

tks :)

**Comment by ncw** [comment]
> i have tried with `--multi-thread-streams 0` , it seems work .

Good!

> Is there any other way？Will the 1.54 release version solve this problem？

It will unless we (or most likely @buengese ) comes up with a fix!

I managed to reproduce this quite easily

```
cd /tmp/
# Make a 1GB ish compressible file
for i in `seq 1000`; do cat ~/go/src/github.com/rclone/rclone/MANUAL.md  >> compressible ; done
mkdir /tmp/compress
# Copy it to a compress remote
rclone copy -vv compressible --compress-remote /tmp/compress :compress:
# Try to recover it
rclone copy -vv --compress-remote /tmp/compress :compress:compressible recovered --low-level-retries 1 --retries 1
```

We get

```
$ rclone copy -vv --compress-remote /tmp/compress :compress:compressible recovered --low-level-retries 1 --retries 1
2021/01/28 17:33:57 DEBUG : rclone: Version "v1.54.0-beta.5106.ebac78a2f.fix-azureblob-memory" starting with parameters ["rclone" "copy" "-vv" "--compress-remote" "/tmp/compress" ":compress:compressible" "recovered" "--low-level-retries" "1" "--retries" "1"]
2021/01/28 17:33:57 DEBUG : Creating backend with remote ":compress:compressible"
2021/01/28 17:33:57 DEBUG : Using config file from "/home/ncw/.rclone.conf"
2021/01/28 17:33:57 DEBUG : Creating backend with remote "recovered"
2021/01/28 17:33:57 DEBUG : fs cache: renaming cache item "recovered" to be canonical "/tmp/recovered"
2021/01/28 17:33:57 DEBUG : compressible: Sizes differ (src 1038532000 vs dst 32768)
2021/01/28 17:33:57 DEBUG : compressible: Starting multi-thread copy with 3 parts of size 330.188M
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 2/3 (346226688-692453376) size 330.188M starting
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.RangeSeek from -1 to 94150702 length -1
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 3/3 (692453376-1038532000) size 330.046M starting
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.RangeSeek from -1 to 188309864 length -1
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.Read at -1 length 4096 chunkOffset 94150702 chunkSize 262144
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.openRange at 94150702 length 262144
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 1/3 (0-346226688) size 330.188M starting
2021/01/28 17:33:57 DEBUG : compressible: Reopening on read failure after 0 bytes: retry 1/1: flate: corrupt input before offset 103
2021/01/28 17:33:57 DEBUG : compressible: Reopen failed after 0 bytes read: failed to reopen: too many retries
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 2/3 failed: multipart copy: read failed: flate: corrupt input before offset 103
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 3/3 failed: context canceled
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.Read at -1 length 4096 chunkOffset 0 chunkSize 262144
2021/01/28 17:33:57 DEBUG : compressible.oL3mPQAAAAA.gz: ChunkedReader.openRange at 0 length 262144
2021/01/28 17:33:57 DEBUG : compressible: multi-thread copy: stream 1/3 failed: context canceled
2021/01/28 17:33:57 ERROR : compressible: Failed to copy: multipart copy: read failed: flate: corrupt input before offset 103
2021/01/28 17:33:57 ERROR : Attempt 1/1 failed with 1 errors and: multipart copy: read failed: flate: corrupt input before offset 103
2021/01/28 17:33:57 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         0.0s

2021/01/28 17:33:57 DEBUG : 4 go routines active
2021/01/28 17:33:57 Failed to copy: multipart copy: read failed: flate: corrupt input before offset 103
```

Whereas

```
rclone copy -vv --compress-remote /tmp/compress :compress:compressible recovered --low-level-retries 1 --retries 1 --multi-thread-streams 0
```

Works perfectly...

**Comment by wangshuang0220** [comment]
got it,  will v1.54 release in Feb.?

thanks a lot

**Comment by buengese** [comment]
Looks like there is indeed something very wrong here. I'll probably fix it this weekend so this will go into 1.54.
